components/accelerator/nvidia/error/xid/component.go

```diff
@@ -17,18 +17,73 @@ import (
 	"github.com/leptonai/gpud/log"
 )
 
+var xidComponent *component
+
 func New(ctx context.Context, cfg Config) components.Component {
 	cfg.Query.SetDefaultsIfNotSet()
 	setDefaultPoller(cfg)
 
 	cctx, ccancel := context.WithCancel(ctx)
 	getDefaultPoller().Start(cctx, cfg.Query, nvidia_component_error_xid_id.Name)
-
-	return &component{
+	// 1. create xid db
+	// 2. create local xid cache
+	// 2.1 warmup cache from db
+	// 3 watcher := dmesg_pkg.NewWatcher()
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				panic("context cancelled")
+			case rawDmesgLine, err := <-watcher.Watch():
+				// 1. parse xid
+				// 2. if not xid, skip
+				// 3. generate current event
+				//   3.1 fill in dmesg based xid
+				//   3.2 fill in nvml based xid (from current poller)
+				// 4. check local cache with timestamp/xid, if it is already in cache, skip
+				// 5. insert into db, cache
```

Comment:
Once we got the latest event, we can insert the event to the event list, and call the function above commented to get the current state.